### ISSUE TITLE
UiSelector being deprecated

### ISSUE THREAD

**Issue by itkhanz** [appium/appium-uiautomator2-driver#812]
Hi,
I wanted to bring this into your attention that UiSelector is being deprecated in future releases for Android UiAutomator2.
Please see https://developer.android.com/training/testing/other-components/ui-automator

![Screenshot 2024-08-07 at 15 40 00](https://github.com/user-attachments/assets/91da1e1a-d5b2-47ab-ae96-eec9fe2ea90f)

If this is true, does that mean any locators using UiSelector will not work in future?

If yes, then would be good to also mark it deprecated soon in Appium so users can adapt their code bases accordingly with alternate locator strategies. Moreover to mark this upcoming deprecation here in docs as well and provide some alternate examples:-
https://github.com/appium/appium-uiautomator2-driver/blob/master/docs/uiautomator-uiselector.md

Moving forward, what would be the recommendation from Appium team to build android locators? Does Appium support building locators via other APIs such as with [BySelector](https://developer.android.com/reference/androidx/test/uiautomator/BySelector)?

See also [#13670](https://github.com/appium/appium/issues/13670)

**Comment by mykola-mokhnach** [comment]
Right now it is probably too early to discuss the deprecation topic, although it definitely makes sense to add the above link to the driver documentation now. Eventually we will have no choice in following Google when they decide to UiObject completely.

cc @KazuCocoa @jlipps

**Comment by jlipps** [comment]
Yeah, good point. Maybe we can create a BySelector parser as we did for UiObject?

**Comment by mitsukuri** [comment]
I have some tests which use UIScrollable selectors failing with Appium `2.12.1`/uiautomator2 `3.9.5` on Browserstack, but running fine on local emulator and physical device. I wonder how is that possible and if that has to do with the deprecation in question..

**Comment by itkhanz** [comment]
> I have some tests which use `UiScrollable` selectors failing with `Appium 2.12.1`/`uiautomator2 3.9.5` on Browserstack, but running fine on my local emulator and physical device. I wonder how is that possible and if that has to do with the deprecation in question..

I don't think this has to do with this deprecation. Please check the UiAutomator2 version on BrowserStack and use the same version to verify whether your tests are running fine locally. Reach out to BrowserStack support.

**Comment by mitsukuri** [comment]
@itkhanz Well it turns out that somewhere, sometime, it became mandatory to use explicit `new UiSelector().scrollable(true).scrollTextIntoView()` in selectors, while it used to work fine with `new UiSelector().scrollTextIntoView()` before.

**Comment by Bonitoflakes** [comment]
Any timeline or plan for the BySelector parser?

**Comment by bhavin-android** [comment]
any one have use appium for jetpack compose ? can you please share sample code for test using appium for compose ?
